.github/workflows/run-pytest.yml

```
@@ -20,12 +20,37 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest, windows-latest, macos-latest]
-        config: [ {python: '3.9', dependencies: 'newest'},
-                  {python: '3.10', dependencies: 'newest'},
-                  {python: '3.11', dependencies: 'newest'},
-                  {python: '3.11', dependencies: 'minimal'},
-                  {python: '3.8', dependencies: 'oldest'} ]
+        os: [ubuntu-latest]
+        python: ['3.9', '3.10', '3.11', '3.12']
```

```suggestion
        python: ['3.8', '3.9', '3.10', '3.11', '3.12']
```
3.8 is needed here - otherwise line 29 and 30 create a new matrix entry without an OS.